File: java/src/days/Day.java
package src.days;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public abstract class Day {
    public abstract String process(String input, boolean isPart2);

    protected String[] parseInputByLines(String input) {
        return input.split("\\r?\\n");
    }

    protected String readInput(String filename) throws IOException {
        Path path = Path.of("resources", filename);
        return Files.readString(path);
    }

    // try {
    //     String input = readInput(inputFileName);
    //     String[] lines = parseInputByLines(input);
    // } catch (IOException e) {
    //     System.err.println("Error reading input file: " + e.getMessage());
    // }
}

File: python/src/days/day.py
from pathlib import Path

class Day:
    def process(self, input_data, is_part_two = False):
        raise NotImplementedError("Subclasses must implement the `process` method!")

    def parse_input_by_lines(self, input_data):
        return input_data.strip().split("\n")

    def read_input(self, filename):
        filepath = Path("resources") / filename
        with open(filepath, "r") as file:
            return file.read()
        